### DISCUSSION TITLE
LIBGL_ALWAYS_SOFTWARE

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#6368] [General]
Hello,

Using Webots headless, I'm using xfvs-run. It run fine, but don't use the GPU, and use software library instead. That not a big deal, but it turn out that Lidar data produced by the Software library are different from those generated using a GPU.

Thus, simulation are not behaving the same way using both methods.

One way to improve thing is to start webots using `LIBGL_ALWAYS_SOFTWARE=1 webots <additionnal parameters>`

It will of course reduce performance, but you end up with simulation behaving much closer that the one executed under xvfs-run

I think it might be useful to add that to the documentation.

**Comment by omichel** [comment]
Do you observe big differences in the lidar data?
If so, yes, please go ahead and propose an improvement of the documentation.
I believe that more generally, small differences may be observed depending on OpenGL renderer (Software/Mesa, NVIDIA, AMD, Intel, etc.).

**Reply by ShuffleWire** [reply]
It's not huge, about 10^-7 m (0.001mm) so that might be very well acceptable
